cardano-api/src/Cardano/Api/MetaData.hs

```diff
@@ -78,44 +78,35 @@ validateTxMetadata
   :: TxMetadata
   -> Either TxMetadataValidationError TxMetadata
 validateTxMetadata txMd@(TxMetadata (MetaData mdMap)) =
-    Map.foldl' foldFn (Right txMd) mdMap
+    txMd <$ traverse_ validate (Map.elems mdMap)
```

Comment:
Previously `txMd` was threaded through the implementation of the function.  I modified the code to use `txMd <$` to indicate that if validation succeeds, the value returned is always `txMd`.  Written this way it's clearer that `validateTxMetadata` will not succeed with any other value.

Reply:
On the RHS of `<$`, we are in `Either` monad, so `traverse_ validate` means to run `validate` on every element in the list, and abort on the first failure.